src/arg_complete.cppo.ml

```diff
@@ -75,6 +75,20 @@ let empty_all _ = []
 type anon_complete = complete
 
 let complete_argv (argv: string list) (speclist: speclist) (anon_complete: complete): string list =
+  (* support autocompletion of -flag=val too. To do this, the minimal edit seems to be filtering out those `=` *)
+  let argv =
+    let rec filter_equals new_args is_last_arg_flag = function
+      | [] -> List.rev new_args
+      | arg :: args ->
+        if Util.starts_with ~prefix:"-" arg then
+          filter_equals (arg :: new_args) true args
+        else if arg = "=" && is_last_arg_flag then
```

Comment:
This seems to be matching when an entire argument is `=`, but isn't `-flag=val` a single argument? Or am I missing something here?

I opened #2 with my alternative attempt to do it similarly to `Stdlib.Arg` by splitting the argument on `=`.

Reply:
After unit testing #2 and everything, I decided to try it out for real and realized the odd things happening here. Apparently bash has some great "features": https://stackoverflow.com/a/57437406/854540, https://stackoverflow.com/a/58441078/854540, https://stackoverflow.com/a/28647824/854540.

Reply:
Yes, I tried #2 and it seems not to work currently. My impression when developing this fix was that somewhere in the middle, something is already splitting `-flag=` into `-flag` and `=`.

All I would like is to find some kind of support for the basic cases, despite the nice "features" of bash :upside_down_face: